src/ngx_http_lua_pipe.c

```diff
@@ -1174,6 +1197,11 @@ ngx_http_lua_pipe_put_error(ngx_http_lua_pipe_ctx_t *pipe_ctx, u_char *errbuf,
 
     case PIPE_ERR_CLOSED:
         *errbuf_size = ngx_snprintf(errbuf, *errbuf_size, "closed") - errbuf;
+        if (pipe_ctx->c != NULL) {
+            ngx_close_connection(pipe_ctx->c);
+            pipe_ctx->c = NULL;
+        }
```

Comment:
I'm worried that this may break the full-duplex feature.

Reply:
Maybe we should make sure both directions are closed before we proactively close and free up the pipe? We can mark this as a TODO for now (and create a github issue to record this?)